src/faultlog/faultlog_main.cpp

```diff
@@ -304,9 +304,12 @@ int main(int argc, char** argv)
             nlohmann::json errorlog = json::array();
             (void)GuardWithEidRecords::populate(bus, unresolvedRecords,
                                                 errorlog);
-            nlohmann::json jsonServiceEvent;
-            jsonServiceEvent["SERVICEABLE_EVENT"] = std::move(errorlog);
-            faultLogJson.emplace_back(jsonServiceEvent);
+            if (!errorlog.empty())
```

Comment:
Suggestion: As the function is large!!
As it is the same piece of code, if we could somehow refactor and create a method, which will take inputs like withEid, unResolvedPels, listFaultLogs as enums and return faultLogJson. Just a thought. I am ok with the current changes

Reply:
Agreed with @deepakala-k . In that way repetition of code and chances of errors would be minimum.

Reply:
void addServiceableEvents(const nlohmann::json& errorlog,
                               nlohmann::json& faultlog)
{
	if(!errorlog.empty())
	{
		nlohmann::json jsonServiceEvent;
		jsonServiceEvent["SERVICEABLE_EVENT"] = std::move(errorlog);
		faultLogJson.emplace_back(jsonServiceEvent);
	}
}

may be change like this